### PR TITLE
Random equation

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### Reference Issues/PRs
#32 
#### What does this implement/fix? Explain your changes.
- Random equations mode added
- Test system updated

### CODE REVIEW

**Comment by sepandhaghighi** [comment]
@sadrasabouri Please review this PR 💯

File: samila/functions.py [hunk 1]
```diff
@@ -4,11 +4,31 @@
...
+from .params import ELEMENTS_LIST, ARGUMENTS_LIST, OPERATORS_LIST
 from .errors import samilaDataError
 
+def random_equation_gen():
+    """
+    Generate random equation.
+
+    :return: equation as str
+    """
+    num_elements = random.randint(2,len(ELEMENTS_LIST) + 3)
```
Comment (sadrasabouri) [inline samila/functions.py:20]: `num_elements = random.randint(2, len(ELEMENTS_LIST) + 3)`

File: samila/functions.py [hunk 2]
```diff
@@ -4,11 +4,31 @@
...
+    result = ""
+    index = 1
+    random_coef = "random.uniform(-1,1)"
+    while(index<=num_elements):
```
Comment (sadrasabouri) [inline samila/functions.py:24]: `index <= num_elements`

File: samila/params.py [hunk 3]
```diff
@@ -50,3 +50,10 @@ class Projection(Enum):
     LAMBERT = "lambert"
     MOLLWEIDE = "mollweide"
     RECTILINEAR = "rectilinear"
+
+
+ELEMENTS_LIST = ["{0}*math.cos({1})","{0}*math.sin({1})","{0}*{1}","{0}*abs({1})","{0}*math.ceil({1})","{0}*math.floor({1})"]
```
Comment (sadrasabouri) [inline samila/params.py:55]: I think it's better if we make it this way:
```

ELEMENTS_LIST = ["{0} * math.cos({1})",
                 "{0} * math.sin({1})",
                 "{0} * {1}",
                 "{0} * abs({1})",
                 "{0} * math.ceil({1})",
                 "{0} * math.floor({1})"]
```
This way we can track further function additions.

File: samila/params.py [hunk 4]
```diff
@@ -50,3 +50,10 @@ class Projection(Enum):
...
+
+ARGUMENTS_LIST = ["x*y","x","y","y-x","x-y","x+y","x**2","y**2","(x**2)*y","(y**2)*x","(x**2)*(y**3)","(x**3)*(y**2)","x*(y**3)","y*(x**3)"]
```
Comment (sadrasabouri) [inline samila/params.py:57]: Also:
```
ARGUMENTS_LIST = ["x * y",
                  "x",
                  "y",
                  "y - x",
                  "x - y",
                  "x + y",
                  "x**2",
                  "y**2",
                  "(x**2) * y",
                  "(y**2) * x",
                  "(x**2) * (y**3)",
                  "(x**3) * (y**2)",
                  "x * (y**3)",
                  "y * (x**3)"]
```

File: samila/params.py [hunk 5]
```diff
@@ -50,3 +50,10 @@ class Projection(Enum):
...
+
+OPERATORS_LIST = ["+","-"]
```
Comment (sadrasabouri) [inline samila/params.py:59]: `OPERATORS_LIST = ["+", "-"]`

File: test/warning_test.py [hunk 6]
```diff
@@ -4,7 +4,7 @@
 >>> import json
 >>> from samila import *
 >>> from pytest import warns
->>> with warns(RuntimeWarning, match="Neither function nor data is provided."):
+>>> with warns(RuntimeWarning, match="Neither function nor data is provided. Samila will switch to random equations mode."):
```
Comment (sadrasabouri) [inline test/warning_test.py:7]: I think we don't need to raise warning anymore. Till now not providing data or function meant to be a warning but now the only way by which user can use the random mode is not providing them (and maybe the most usable feature of Samila) so I think we don't need this warning anymore.

File: samila/functions.py [hunk 7]
```diff
@@ -4,11 +4,31 @@
...
+        argument = random.choice(ARGUMENTS_LIST)
+        result = result + random.choice(ELEMENTS_LIST).format(random_coef,argument)
```
Comment (sadrasabouri) [inline samila/functions.py:26]: `format(random_coef, argument)`

File: samila/functions.py [hunk 8]
```diff
@@ -4,11 +4,31 @@
...
+        if index<num_elements:
```
Comment (sadrasabouri) [inline samila/functions.py:27]: `if index < num_elements:`

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Thanks for your effort.
I left some comments about styles and also a change request.

File: samila/genimage.py [hunk 9]
```diff
@@ -35,11 +35,15 @@ def __init__(self, function1=None, function2=None, data=None):
         """
         self.function1 = function1
         self.function2 = function2
+        self.function1_str = None
+        self.function2_str = None
```
Comment (sadrasabouri) [inline samila/genimage.py:39]: Log these new attributes into `CHANGELOG.md` too.

File: README.md [hunk 10]
```diff
@@ -99,12 +99,28 @@ Samila is a generative art generator written in Python, Samila let's you create
 
 ## Usage
 
+### Magic
```
Comment (sadrasabouri) [inline README.md:102]: What about `Random`?
We can also don't use any header here as our main method to use Samila is this.

**Review by sadrasabouri — COMMENTED** [review]
Also I still think if we add a space between operator and its operands (expect `**`) it'll look like better (but surly longer).

File: README.md (hunk 10 above)
Comment (sepandhaghighi) [reply README.md:102]: I think `magic` is better, it is more fancy

File: samila/genimage.py (hunk 9 above)
Comment (sepandhaghighi) [reply samila/genimage.py:39]: Done!

**Review by sadrasabouri — APPROVED** [review]
All good.

**Pull request merged**